lib/features/debug/presentation/screens/debug/debug_model.dart

```diff
@@ -1,62 +1,50 @@
-import 'package:flutter/material.dart';
-import 'package:flutter_template/common/service/theme/theme_service.dart';
-import 'package:flutter_template/config/app_config.dart';
+import 'package:flutter/foundation.dart';
 import 'package:flutter_template/config/url.dart';
 import 'package:flutter_template/core/architecture/presentation/base_model.dart';
 import 'package:flutter_template/features/debug/domain/repositories/i_debug_repository.dart';
 import 'package:flutter_template/features/debug/presentation/screens/debug/debug_screen.dart';
 
+/// {@template debug_model.class}
 /// Model for [DebugScreen].
+/// {@endtemplate}
 final class DebugScreenModel extends BaseModel {
   final IDebugRepository _debugRepository;
-  final AppConfig _appConfig;
-  final IThemeService _themeService;
 
-  /// Config change Notifier.
-  late ValueNotifier<AppConfig> configNotifier;
-
-  /// Theme mode change Notifier.
-  late ValueNotifier<ThemeMode> currentThemeMode;
-
-  /// Config proxy url value.
-  String? get proxyUrl => _appConfig.proxyUrl;
-
-  /// Create an instance [DebugScreenModel].
-  DebugScreenModel(
-    this._debugRepository,
-    this._appConfig,
-    this._themeService, {
+  /// {@macro debug_model.class}
+  DebugScreenModel({
+    required IDebugRepository debugRepository,
+    required Url serverUrl,
     required super.logWriter,
-  });
-
-  @override
-  void init() {
-    configNotifier = ValueNotifier<AppConfig>(_appConfig);
-    currentThemeMode = ValueNotifier<ThemeMode>(_themeService.currentThemeMode);
-    _themeService.addListener(_updateTheme);
+  }) : _debugRepository = debugRepository {
+    _serverUrlState = ValueNotifier(serverUrl);
   }
 
-  @override
-  void dispose() {
-    _themeService.removeListener(_updateTheme);
-  }
+  late final ValueNotifier<Url> _serverUrlState;
+
+  /// Current server url state.
+  ValueListenable<Url> get serverUrlState => _serverUrlState;
```

Comment:
As far as I can see it is pure UI state. So it should be in `wm`, not in `model`.